www/scripts/productlist/ListOfProducts.js

```diff
@@ -213,15 +214,46 @@ function (declare, domClass, domConstruct, ItemFileWriteStore, topic,
         + name[0].toUpperCase()
         + '</span></div>';
 
-      if (!item.connected || !item.accessible) {
+
+      var dbStatus = item.databaseStatus
+
+      var db_status_text = ""
+
+      switch(dbStatus) {
+        case DBStatus.OK:
+            db_status_text = ""
+            break;
+        case DBStatus.MISSING:
+            db_status_text = "Database is missing."
+            break;
+        case DBStatus.FAILED_TO_CONNECT:
+            db_status_text = "Failed to connect to the database."
+            break;
+        case DBStatus.SCHEMA_MISMATCH_OK_MIGRATION:
+            db_status_text = "Schema mismatch: migration is possible."
+            break;
+        case DBStatus.SCHEMA_MISMATCH_NO_MIGRATION:
+            db_status_text = "Schema mismatch: migration not available."
+            break;
+        case DBStatus.SCHEMA_MISSING:
+            db_status_text = "Schema is missing."
+            break;
+        case DBStatus.SCHEMA_INITIALIZATION_ERROR:
```

Comment:
This is also duplication. I think it is possible to define constants in thrift. Using a map constant from enum to string might reduce the duplication also across languages.